### ISSUE TITLE
generated code refers to a package: import from a src/ directory

### ISSUE THREAD

**Issue by devoncarew** [dart-lang/intl#151]
We're seeing lint issues reported for generated intl code:

```
info: Don't import implementation files from another package. (implementation_imports at [l10n_example] lib/l10n/messages_all.dart:9)
```

(some context: https://github.com/flutter/flutter/issues/12726)

`messages_all.dart` has this import: `import 'package:intl/src/intl_helpers.dart';`.

Generally, so don't want code to reach into src/ directories of other packages, since there's no API stability guarantee for things in src/. Likely not a issue in this specific case, since package:intl both provides the API and generates the code using the API. It would be nice to address however, just the uniformity w/ the style guide.

One possibility is to move `intl_helpers.dart` into a `package:intl/helpers` dir (or some such). You could leave a re-export back in the src/ dir for existing code.

More short term, I may open a PR to suppress this specific warning.

@alan-knight

**Comment by HansMuller** [comment]
It would be helpful if the code generated by intl_translation:generate_from_arb would pass muster with the Flutter analyzer options: https://github.com/flutter/flutter/blob/master/packages/flutter/lib/analysis_options_user.yaml

Developers of i18n apps are likely to `include: package:flutter/analysis_options_user.yaml`. 

Currently the flutter repo excludes the intl-generated code in our one example that uses intl: https://github.com/flutter/flutter/blob/master/analysis_options.yaml#L42

**Comment by devoncarew** [comment]
For those two excludes:

```
- 'lib/i18n/stock_messages_*.dart'
- 'lib/src/http/**'
```

Do you happen to remember what the lints were that were being hit?

**Comment by devoncarew** [comment]
Ah, just one of the excludes is the intl code.

**Comment by devoncarew** [comment]
A fix for this landed internally (in package:intl_translation). @alan-knight, do you mind doing a re-sync from internal, and I assume a re-publish to capture the change?

**Comment by devoncarew** [comment]
OK, so there are more than a few issues when analyzed in strong mode, with the default set of flutter lints:

```
   info • Don't import implementation files from another package. • examples/stocks/lib/i18n/stock_messages_all.dart:9:8
   info • Specify type annotations. • examples/stocks/lib/i18n/stock_messages_all.dart:15:49
  error • Missing type arguments for generic type 'Future<dynamic>'. • examples/stocks/lib/i18n/stock_messages_all.dart:16:19
   info • Specify type annotations. • examples/stocks/lib/i18n/stock_messages_all.dart:16:19
  error • Missing type arguments for generic type 'Future<dynamic>'. • examples/stocks/lib/i18n/stock_messages_all.dart:17:19
   info • Specify type annotations. • examples/stocks/lib/i18n/stock_messages_all.dart:17:19
  error • Missing parameter type for 'localeName'. • examples/stocks/lib/i18n/stock_messages_all.dart:20:35
   info • Specify type annotations. • examples/stocks/lib/i18n/stock_messages_all.dart:20:35
   info • Specify type annotations. • examples/stocks/lib/i18n/stock_messages_all.dart:32:1
   info • Specify type annotations. • examples/stocks/lib/i18n/stock_messages_all.dart:33:3
   info • Prefer final for variable declaration if reference is not reassigned. • examples/stocks/lib/i18n/stock_messages_all.dart:33:7
   info • Specify type annotations. • examples/stocks/lib/i18n/stock_messages_all.dart:34:3
   info • Prefer final for variable declaration if reference is not reassigned. • examples/stocks/lib/i18n/stock_messages_all.dart:34:7
   info • Specify type annotations. • examples/stocks/lib/i18n/stock_messages_all.dart:34:32
  error • Missing type arguments for generic method 'then<S>'. • examples/stocks/lib/i18n/stock_messages_all.dart:35:15
  error • Missing parameter type for '_'. • examples/stocks/lib/i18n/stock_messages_all.dart:35:21
   info • Specify type annotations. • examples/stocks/lib/i18n/stock_messages_all.dart:42:3
  error • Missing variable type for 'messages'. • examples/stocks/lib/i18n/stock_messages_all.dart:42:7
   info • Avoid empty catch blocks. • examples/stocks/lib/i18n/stock_messages_all.dart:45:15
  error • Missing parameter type for 'locale'. • examples/stocks/lib/i18n/stock_messages_all.dart:49:50
   info • Specify type annotations. • examples/stocks/lib/i18n/stock_messages_all.dart:49:50
   info • Specify type annotations. • examples/stocks/lib/i18n/stock_messages_all.dart:50:3
   info • Prefer final for variable declaration if reference is not reassigned. • examples/stocks/lib/i18n/stock_messages_all.dart:50:7
  error • Missing parameter type for '_'. • examples/stocks/lib/i18n/stock_messages_all.dart:51:19
   info • Separate the control structure expression from its statement. • examples/stocks/lib/i18n/stock_messages_all.dart:52:29
   info • Specify type annotations. • examples/stocks/lib/i18n/stock_messages_en.dart:9:1
   info • Specify type annotations. • examples/stocks/lib/i18n/stock_messages_en.dart:11:1
   info • Annotate overridden members • examples/stocks/lib/i18n/stock_messages_en.dart:14:7
   info • Declare method return types. • examples/stocks/lib/i18n/stock_messages_en.dart:14:7
   info • Specify type annotations. • examples/stocks/lib/i18n/stock_messages_en.dart:16:3
   info • Annotate overridden members • examples/stocks/lib/i18n/stock_messages_en.dart:16:9
  error • Missing return type for '_notInlinedMessages'. • examples/stocks/lib/i18n/stock_messages_en.dart:17:3
   info • Declare method return types. • examples/stocks/lib/i18n/stock_messages_en.dart:17:10
  error • Missing parameter type for '_'. • examples/stocks/lib/i18n/stock_messages_en.dart:17:30
  error • Missing type arguments for map literal. • examples/stocks/lib/i18n/stock_messages_en.dart:17:36
   info • Specify type annotations. • examples/stocks/lib/i18n/stock_messages_en.dart:17:36
   info • Prefer single quotes where they won't require escape sequences • examples/stocks/lib/i18n/stock_messages_en.dart:18:5
   info • Prefer single quotes where they won't require escape sequences • examples/stocks/lib/i18n/stock_messages_en.dart:18:53
   info • Prefer single quotes where they won't require escape sequences • examples/stocks/lib/i18n/stock_messages_en.dart:19:5
   info • Prefer single quotes where they won't require escape sequences • examples/stocks/lib/i18n/stock_messages_en.dart:19:56
   info • Prefer single quotes where they won't require escape sequences • examples/stocks/lib/i18n/stock_messages_en.dart:20:5
   info • Prefer single quotes where they won't require escape sequences • examples/stocks/lib/i18n/stock_messages_en.dart:20:52
   info • Specify type annotations. • examples/stocks/lib/i18n/stock_messages_es.dart:9:1
   info • Specify type annotations. • examples/stocks/lib/i18n/stock_messages_es.dart:11:1
   info • Annotate overridden members • examples/stocks/lib/i18n/stock_messages_es.dart:14:7
   info • Declare method return types. • examples/stocks/lib/i18n/stock_messages_es.dart:14:7
   info • Specify type annotations. • examples/stocks/lib/i18n/stock_messages_es.dart:16:3
   info • Annotate overridden members • examples/stocks/lib/i18n/stock_messages_es.dart:16:9
  error • Missing return type for '_notInlinedMessages'. • examples/stocks/lib/i18n/stock_messages_es.dart:17:3
   info • Declare method return types. • examples/stocks/lib/i18n/stock_messages_es.dart:17:10
  error • Missing parameter type for '_'. • examples/stocks/lib/i18n/stock_messages_es.dart:17:30
  error • Missing type arguments for map literal. • examples/stocks/lib/i18n/stock_messages_es.dart:17:36
   info • Specify type annotations. • examples/stocks/lib/i18n/stock_messages_es.dart:17:36
   info • Prefer single quotes where they won't require escape sequences • examples/stocks/lib/i18n/stock_messages_es.dart:18:5
   info • Prefer single quotes where they won't require escape sequences • examples/stocks/lib/i18n/stock_messages_es.dart:18:53
   info • Prefer single quotes where they won't require escape sequences • examples/stocks/lib/i18n/stock_messages_es.dart:19:5
   info • Prefer single quotes where they won't require escape sequences • examples/stocks/lib/i18n/stock_messages_es.dart:19:56
   info • Prefer single quotes where they won't require escape sequences • examples/stocks/lib/i18n/stock_messages_es.dart:20:5
   info • Prefer single quotes where they won't require escape sequences • examples/stocks/lib/i18n/stock_messages_es.dart:20:52
59 issues found • analyzed 311 files, 11.07 seconds
```

**Comment by alan-knight** [comment]
Published a 0.16.1 of intl with this change, and also fixed the other lints that were coming up for me from a flutter default set.
